response_operations_ui/forms.py

```diff
@@ -451,29 +451,27 @@ def validate_email(_, field):
 
 
 class SetAccountPasswordForm(FlaskForm):
+    # Password match check has been removed from password validator as it is redundant (main password dictates match).
+    # Regex is used to check password has correct characters. This also includes a check on length range.
+    # Redundant range validator removed. Due to the inclusion of regexp, the validate_password method has been removed.
+    # Password confirm only runs a check to ensure it matches with password. No additional checks required.
+
+    regex_formula = r"^(?=[A-Za-z0-9\W])^(?=.*[a-z])(?=.*[A-Z])(?=.*[\W])(?=.*[0-9])(?=.{12,160}).*$"
```

Comment:
If you're going to add a long and complicated regex like that then it 100% needs to have some very descriptive unit tests to go with it to describe what is and isn't allowed.  Can you write some tests that really exercise what are and aren't valid passwords.